pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -51,14 +49,25 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 	clusters := make([]*v2.Cluster, 0)
 
 	services := push.Services
+	recomputeOutboundClusters := true
 
-	clusters = append(clusters, configgen.buildOutboundClusters(env, proxy, push, services)...)
-	for _, c := range clusters {
-		// Envoy requires a non-zero connect timeout
-		if c.ConnectTimeout == 0 {
-			c.ConnectTimeout = defaultClusterConnectTimeout
+	switch proxy.Type {
+	case model.Sidecar:
+		if configgen.OutboundClustersForSidecars != nil {
+			clusters = append(clusters, configgen.OutboundClustersForSidecars...)
+			recomputeOutboundClusters = false
+		}
+	case model.Router:
+		if configgen.OutboundClustersForGateways != nil {
```

Comment:
Do all routers/gatway have same clusters ? No source-based config ?

Reply:
nope.. All same

Reply:
And even this differentiation will go away if we get rid of the global MTLS parsing thing which was not supposed to be available in Istio 1.0.